File: src/main/java/javabot/dao/util/CleanHtmlConverter.java
package javabot.dao.util;

import java.util.StringJoiner;
import java.util.function.Function;
import java.util.stream.Stream;

public class CleanHtmlConverter {
    public static String convert(String message, Function<String, String> converter) {
        message = message.replaceAll("&", "&amp;")
                .replaceAll("<", "&lt;")
                .replaceAll(">", "&gt;");
        if (message.contains("http://") || message.contains("https://")) {
            StringJoiner joiner = new StringJoiner(" ");
            Stream.of(message.split(" ")).map(token -> {
                if (token.startsWith("http://") || token.startsWith("https://")) {
                    token = converter.apply(token);
                }
                return token;
            }).forEach(joiner::add);

            message = joiner.toString();
        }
        return message;
    }
}

File: src/main/kotlin/javabot/dao/util/CleanHtmlConverter.kt
package javabot.dao.util

object CleanHtmlConverter {
    fun convert(message: String, converter: (String) -> String): String {
        var content = message.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")

        if (content.contains("http://") || content.contains("https://")) {
            content =
                content
                    .split(" ")
                    .map({ token ->
                        var value = token
                        if (token.startsWith("http://") || token.startsWith("https://")) {
                            value = converter(token)
                        }
                        value
                    })
                    .joinToString(" ")
        }
        return content
    }
}
